### DISCUSSION TITLE
Accessing optional values in array-like structures

### DISCUSSION THREAD

**Discussion by janoschherrmann** [amannn/next-intl#857] [Q&A] [answered]
Say I have an array of objects like this:
`type MyObject = {
  title: string;
  optionalProperty?: string
};` 
I want to have add this object into my messages. Now in my code, I'd like to map over these different objects (using the mechanism described [in the docs](https://next-intl-docs.vercel.app/docs/usage/messages#arrays-of-messages)). When I try to access `optionalProperty`, I would get a type error. Is there a way to prevent this? Could I maybe check whether a certain value exists, and if not, fallback to some other value?

**Comment by amannn** [comment, marked as answer]
This is discussed here: https://github.com/amannn/next-intl/issues/88

**Reply by janoschherrmann** [reply]
Ah thanks, didn't find that thread before :)
